dataflow/operators/comtrade.py

```
@@ -0,0 +1,138 @@
+import codecs
+import csv
+import io
+import json
+import zipfile
+
+import backoff
+import requests
+
+from dataflow import config
+from dataflow.utils import S3Data, logger
+
+
+def fetch_comtrade_goods_data(
+    table_name: str, ts_nodash: str, **kwargs,
+):
+    s3 = S3Data(table_name, ts_nodash)
+    expected_keys = [
+        'Classification',
+        'Year',
+        'Period',
+        'Period Desc.',
+        'Aggregate Level',
+        'Is Leaf Code',
+        'Trade Flow Code',
+        'Trade Flow',
+        'Reporter Code',
+        'Reporter',
+        'Reporter ISO',
+        'Partner Code',
+        'Partner',
+        'Partner ISO',
+        'Commodity Code',
+        'Commodity',
+        'Qty Unit Code',
+        'Qty Unit',
+        'Qty',
+        'Netweight (kg)',
+        'Trade Value (US$)',
+        'Flag',
+    ]
+    _fetch(s3, 'C', expected_keys)
+
+
+def fetch_comtrade_services_data(
+    table_name: str, ts_nodash: str, **kwargs,
+):
+    s3 = S3Data(table_name, ts_nodash)
+    expected_keys = [
+        'Classification',
+        'Year',
+        'Period',
+        'Period Desc.',
+        'Aggregate Level',
+        'Is Leaf Code',
+        'Trade Flow Code',
+        'Trade Flow',
+        'Reporter Code',
+        'Reporter',
+        'Reporter ISO',
+        'Partner Code',
+        'Partner',
+        'Partner ISO',
+        'Commodity Code',
+        'Commodity',
+        'Trade Value (US$)',
+        'Flag',
+    ]
+    _fetch(s3, 'S', expected_keys)
+
+
+def _fetch(s3, trade_type, expected_keys):
+    with _download('https://comtrade.un.org/Data/cache/years.json') as response:
+        years_objs = json.loads(response.content)
+    if years_objs['more']:
+        raise Exception('Unable to fetch all years')
```

Do we never expect this key to appear? This could probably use either a comment explaining why it's a hard error or just handling the edge(?) case?

---

I think this can be removed. I suppose the `more` key will be set to true when there is another page of years but I can't imagine that happening any time soon.

---

Ah ok. Maybe it is better to leave it to hard error in the future rather than silently ignore additional years...